deno/gateway/v10.ts

```diff
@@ -879,15 +879,17 @@ export interface GatewayGuildCreateDispatchData extends APIGuild {
 	 *
 	 * See https://discord.com/developers/docs/resources/channel#channel-object
 	 */
-	channels: APIChannel[];
+	channels: (APIChannel & { type: Exclude<ChannelType, ChannelType.DM | ChannelType.GroupDM> })[];
```

Comment:
[GuildChannelType](https://github.com/discordjs/discord-api-types/blob/main/payloads/v10/channel.ts#L55) can be used here